From 417b6eb0f09dc73984a7dba2aa42c9d8683d5294 Mon Sep 17 00:00:00 2001
From: Jean-Baptiste Kempf <jb@videolan.org>
Date: Sat, 11 Oct 2014 17:53:45 +0200
Subject: [PATCH] Win32: prefer the static libraries when creating the dlls

This should avoid the issues of packaging libgcc_s_sjlj-1.dll,
libwinpthread-1.dll and the like.
---
 lib/Makefile.am   |    3 ++-
 modules/common.am |    3 ++-
 src/Makefile.am   |    3 ++-
 3 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/lib/Makefile.am b/lib/Makefile.am
index c53fe08..be1b6be 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -64,7 +64,8 @@ libvlc_la_LDFLAGS = \
 libvlc_la_DEPENDENCIES = libvlc.sym
 if HAVE_WIN32
 libvlc_la_DEPENDENCIES += ../src/libvlc_win32_rc.$(OBJEXT)
-libvlc_la_LDFLAGS += -Wl,../src/libvlc_win32_rc.$(OBJEXT) -avoid-version
+libvlc_la_LDFLAGS += -Wl,../src/libvlc_win32_rc.$(OBJEXT) -avoid-version \
+					 -Wc,--static -Wc,-static-libgcc
 endif
 if HAVE_OS2
 libvlc_la_LDFLAGS += -avoid-version
diff --git a/modules/common.am b/modules/common.am
index 9bb5143..c3b3f8e 100644
--- a/modules/common.am
+++ b/modules/common.am
@@ -33,7 +33,8 @@ AM_LDFLAGS = \
 	-no-undefined \
 	$(top_builddir)/compat/libcompat.la $(LTLIBVLCCORE) 
 if HAVE_WIN32
-AM_LDFLAGS += $(top_builddir)/modules/module.rc.lo
+AM_LDFLAGS += $(top_builddir)/modules/module.rc.lo \
+              -Wc,--static -Wc,-static-libgcc
 endif
 
 SUFFIXES = .asm
diff --git a/src/Makefile.am b/src/Makefile.am
index 1e113c9..4472271 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -190,7 +190,8 @@ libvlccore_la_LIBADD = $(LIBS_libvlccore) \
 libvlccore_la_DEPENDENCIES = libvlccore.sym
 if HAVE_WIN32
 libvlccore_la_DEPENDENCIES += libvlc_win32_rc.$(OBJEXT)
-libvlccore_la_LDFLAGS += -Wl,libvlc_win32_rc.$(OBJEXT) -avoid-version
+libvlccore_la_LDFLAGS += -Wl,libvlc_win32_rc.$(OBJEXT) -avoid-version \
+						 -Wc,--static -Wc,-static-libgcc
 endif
 if HAVE_OS2
 libvlccore_la_LDFLAGS += -avoid-version
-- 
1.7.10.4
